squire_lib/src/ffi_rounds.rs

```diff
@@ -0,0 +1,160 @@
+use crate::ffi::FFI_TOURNAMENT_REGISTRY;
+use crate::{
+    identifiers::{PlayerId, RoundId, TournamentId},
+    round::{Round, RoundStatus},
+};
+use std::alloc::{Allocator, Layout, System};
+use std::collections::HashSet;
+use std::option::Option;
+use std::ptr;
+use uuid::Uuid;
+
+impl RoundId {
+    /// Returns the round if it can be found in the tournament
+    fn get_tourn_round(self, tid: TournamentId) -> Option<Round> {
+        unsafe {
+            match FFI_TOURNAMENT_REGISTRY.get_mut().unwrap().get(&tid) {
+                Some(t) => {
+                    return t.round_reg.get_round(&self.into()).cloned();
```

Comment:
I know the player has this clone too, but it really isn't needed, right? The `DashMap` ensures mutual exclusion, so we are good to just return a reference to the object since these objects aren't crossing the language boundary.

Reply:
I'm getting errors without cloned about it return Option<&Round>

Reply:
What are those errors? Is it a lifetime error?

Reply:
I think so